code/figure6.jl

```diff
@@ -0,0 +1,68 @@
+include("seed.jl")
+
+# Simulation with specialist natural enemies with stochasticity on D
+sim1, params1 = simulation(50.0, 25.0, m=0.5, F=4.0, D=0.5, h=10.0, b=25.0, a=0.5, th= 0.0, m=0.5, f=generalist_dyn, D_sd=0.5)
+# Fig 5a)
+fig6a = plot(sim1[:,1],sim1[:,2], label="Hosts", frame=:origin, lw=3, leg=false, ylims=(0, 90))
+plot!(fig6a, sim1[:,1],sim1[:,3], label="Parasites", lw=3)
+ylabel!(fig6a, "Population size")
+annotate!(fig6a, 5, 85, text("(a)"))
+annotate!(fig6a, 25, 80, text("D = 0.5 ± 0.5"))
+
+kval1 = mapslices((r) -> kvalue_by_generation(r, params1), sim1, 2)
+fig6d = plot(log10.(sim1[:,2]), kval1,
+   m=:circle, msc=:white, msw=2, ms=4, mc=:grey,
+   lc=:grey, lw=2,
+   leg=false, frame=:origin,
+   xlims=(-0.5, 2.0),
+   ylims=(0, 1.5))
+ylabel!(fig6d, "k-value")
+annotate!(fig6d, -0.25, 1.35, text("(d)"))
+
+sim1_R2 = cor(vec(log10.(sim1[:,2])), vec(kval1))^2
+sim1_a, sim1_b = linreg(vec(log10.(sim1[:,2])), vec(kval1))
+
+# Simulation with specialist natural enemies with stochasticity on c
+sim2, params2 = simulation(50.0, 25.0, m=0.5, F=4.0, D=0.5, h=10.0, b=25.0, a=0.5, th= 0.0, m=0.5, f=generalist_dyn, h_sd=5)
+# Fig 5b)
+fig6b = plot(sim2[:,1],sim2[:,2], label="Hosts", frame=:origin, lw=3, leg=false, ylims=(0, 90))
```

Comment:
every plot command is missing the space after the `,` of the first arg.

Reply:
Fixed.